_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -806,6 +806,28 @@ mfxStatus ImplementationAvc::Init(mfxVideoParam * par)
         if (sts != MFX_ERR_NONE)
             return MFX_WRN_PARTIAL_ACCELERATION;
         m_video.mfx.RateControlMethod = storedRateControlMethod;
+
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+        mfxExtCodingOption3 * extOpt3 = GetExtBuffer(m_video);
+        if (IsOn(extOpt3->ExtBrcAdaptiveLTR) &&
+            (m_video.mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_PROGRESSIVE) &&
+            (m_video.mfx.RateControlMethod == MFX_RATECONTROL_CBR || m_video.mfx.RateControlMethod == MFX_RATECONTROL_VBR)) {
+            mfxExtCodingOptionDDI const * extDdi = GetExtBuffer(m_video);
+            mfxU32 numActiveRefL0P = extDdi->NumActiveRefP;
```

Comment:
We have Active refs control in mfxExtCodingOption3:
    mfxU16      NumRefActiveP[8];
    mfxU16      NumRefActiveBL0[8];
    mfxU16      NumRefActiveBL1[8];

I'd recommend redesign to use them instead of undocumented internal buffer

Reply:
Active refs not supported by H.264 HW.

Reply:
Yes :(, thus can be ignored now.